clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -119,6 +145,17 @@ public boolean ready(Node node, long now) {
     }
 
     /**
+     * Disconnect connection to a particular node (if there is one). The disconnection is asynchronous and will not be
+     * processed until the next {@link #poll(long, long) poll()} call.
+     *
+     * @param nodeId The id of the node
+     */
+    @Override
+    public void disconnect(String nodeId) {
+        selector.disconnect(nodeId);
```

Comment:
I did think about setting the state to DISCONNECTED (which has a similar effect), but then I thought that this would be updated automatically via `poll` as per the comment added in the discussion around the javadoc for `KafkaClient.poll`. But maybe the right thing is to do update the state here as you suggest. Do you think we should remove the id or set the state to `DISCONNECTED`?